lib/salestation/web/ip_address.rb

```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+module Salestation
+  class Web < Module
+    module IPAddress
+        include Deterministic
+
+        def self.extract(request)
+          ip_address = if request['HTTP_X_FORWARDED_FOR'].nil?
+             request['REMOTE_ADDR']
+            else
+             request['HTTP_X_FORWARDED_FOR'].split(',').first
+            end
+          Result::Success(ip_address)
```

Comment:
I don't think you ned to wrap it in Result::Success. I think just returning the IP here is fine.